src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/Symbolic/BDDAlgebra.cs

```diff
@@ -74,10 +72,11 @@ internal abstract class BDDAlgebra : IBooleanAlgebra<BDD>
         public BDD Not(BDD a) =>
             a == False ? True :
             a == True ? False :
-            _opCache.GetOrAdd((BoolOp.Not, a, null), static (key, algebra) => key.a.IsLeaf ?
-                algebra.GetOrCreateBDD(algebra.CombineTerminals(BoolOp.Not, key.a.Ordinal, 0), null, null) : // multi-terminal case
-                algebra.GetOrCreateBDD(key.a.Ordinal, algebra.Not(key.a.One), algebra.Not(key.a.Zero)),
-                this);
+            _opCache.GetOrAdd((BoolOp.Not, a, null), static (key, algebra) =>
```

Comment:
No need to do this on your PR, but we should probably later have better parameter names in all of these methods instead of `op` `a` `b` to make the code easier to read.